engine/src/flutter/shell/platform/darwin/ios/framework/Source/SemanticsObject.mm

```diff
@@ -836,6 +836,22 @@ - (UIAccessibilityTraits)accessibilityTraits {
   return traits;
 }
 
+- (BOOL)accessibilityRespondsToUserInteraction {
```

Comment:
@chunhtai I'm worried this logic might break if/when we add more "system" actions. Would you have any recommendations on how to future-proof this method?

For example, should we push `kSystemActions` down to `//flutter/lib/ui/semantics/semantics_node.h`?

Reply:
yeah, that sounds like a better way of keeping track of this.

Reply:
also should we implement this in SemanticsObject class instead? there are subclasses such as switch that extends from SemanticsObject, for example, a disabled switch probability shouldn't responseToUserInteraction as well.

Reply:
This is a good question. I believe `FlutterSwitchSemanticsObject` works as expected as it overrides traits, and I didn't see any issues Material 3 switches. However, I **_think_** we _might_ need a similar fix for `FlutterScrollableSemanticsObject`.

Voice Control shows incorrect numbers on the Material 3 demo, and I suspected `FlutterScrollableSemanticsObject` but I didn't have the time to root cause it. @LouiseHsu could you check if pushing this logic down to `SemanticsObject` fixes the Material 3 demo app?

Reply:
I tried this with our fix, and the numbers did change, but not for the number on the grey bar 
| Before Fix | After Fix |
| ------------- | ------------- |
| ![before_fix](https://github.com/user-attachments/assets/428eb50e-b033-475b-8690-ca625c355580) |![after_fix](https://github.com/user-attachments/assets/303bb5bb-8ed5-42b7-a481-815c7d2e4222) |

Interestingly if i comment out the other sibling containers the marker disappears. I think the root cause is different than the fix for these two issues. 
![just_actions](https://github.com/user-attachments/assets/88343429-44a1-4feb-bd00-09ba6c937e4c)